src/CsrfProtection.php

```diff
@@ -66,12 +66,14 @@ public function validateCookie(WebRequest $request)
     public function getCookie()
     {
         $request = Request::current();
+        $settings = CsrfSettings::singleton();
 
         if ($request instanceof WebRequest){
             $existingCookie = $request->cookie(self::TOKEN_COOKIE_NAME, false);
 
             if ($existingCookie){
                 $this->currentCookie = $existingCookie;
+                HttpResponse::setCookie(self::TOKEN_COOKIE_NAME, $existingCookie, 0, '/', parse_url($settings->domain, PHP_URL_HOST), false, true);
```

Comment:
@acuthbert do you feel I should include this code? I added it to ensure that any previous csrf_tk cookie that was not HTTP_ONLY would be updated but I wanted to clarify with you if you feel it is necessary